examples/chat/client/chat.cc

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "absl/base/thread_annotations.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/escaping.h"
+#include "absl/synchronization/mutex.h"
+#include "asylo/util/logging.h"
+#include "examples/chat/proto/chat.grpc.pb.h"
+#include "examples/chat/proto/chat.pb.h"
+#include "examples/utils/utils.h"
+#include "include/grpcpp/grpcpp.h"
+#include "oak/client/application_client.h"
+#include "oak/client/manager_client.h"
+#include "oak/common/nonce_generator.h"
+
+ABSL_FLAG(std::string, manager_address, "127.0.0.1:8888",
+          "Address of the Oak Manager to connect to");
+ABSL_FLAG(std::string, module, "", "File containing the compiled WebAssembly module");
+ABSL_FLAG(std::string, app_address, "",
+          "Address of the Oak Application to connect to (empty to create a new application)");
+ABSL_FLAG(std::string, room_name, "", "Name of room to create");
+ABSL_FLAG(std::string, room_id, "",
```

Comment:
How about `--room_access_token` ? It makes it more obvious that this is a secret. Also in the log it would read better if you print out something like "generating access token" and then print that out too, it makes the flow clearer (I think). Also if you do this rename the corresponding proto field.

Reply:
Though I'm thinking whether we should keep this as `room_id`, and then additionally pass the token as policy. Anyways, feel free to ignore this for now, we can discuss offline, but let me know if you have any thoughts either way.